seatunnel-connectors-v2/connector-file/connector-file-base/src/main/java/org/apache/seatunnel/connectors/seatunnel/file/sink/writer/JsonWriteStrategy.java

```diff
@@ -0,0 +1,98 @@
+import org.apache.seatunnel.api.table.type.SeaTunnelRowType;
+import org.apache.seatunnel.connectors.seatunnel.file.sink.config.TextFileSinkConfig;
+import org.apache.seatunnel.connectors.seatunnel.file.sink.util.FileSystemUtils;
+import org.apache.seatunnel.format.json.JsonSerializationSchema;
+
+import lombok.NonNull;
+import org.apache.hadoop.fs.FSDataOutputStream;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+
+public class JsonWriteStrategy extends AbstractWriteStrategy {
+    private final byte[] rowDelimiter;
+    private SerializationSchema serializationSchema;
+    private final Map<String, FSDataOutputStream> beingWrittenOutputStream;
+
+    public JsonWriteStrategy(TextFileSinkConfig textFileSinkConfig) {
+        super(textFileSinkConfig);
+        this.beingWrittenOutputStream = new HashMap<>();
+        this.rowDelimiter = textFileSinkConfig.getRowDelimiter().getBytes();
+    }
+
+    @Override
+    public void setSeaTunnelRowTypeInfo(SeaTunnelRowType seaTunnelRowType) {
+        super.setSeaTunnelRowTypeInfo(seaTunnelRowType);
+        this.serializationSchema = new JsonSerializationSchema(seaTunnelRowType);
+    }
+
+    @Override
+    public void write(@NonNull SeaTunnelRow seaTunnelRow) {
+        String filePath = getOrCreateFilePathBeingWritten(seaTunnelRow);
+        FSDataOutputStream fsDataOutputStream = getOrCreateOutputStream(filePath);
+        try {
+            byte[] rowBytes = serializationSchema.serialize(seaTunnelRow);
+            fsDataOutputStream.write(rowBytes);
+            fsDataOutputStream.write(rowDelimiter);
+        } catch (IOException e) {
+            log.error("write data to file {} error", filePath);
+            throw new RuntimeException(e);
+        }
+    }
+
+    @Override
+    public void finishAndCloseFile() {
+        beingWrittenOutputStream.forEach((key, value) -> {
+            try {
+                value.flush();
+            } catch (IOException e) {
+                log.error("error when flush file {}", key);
+                throw new RuntimeException(e);
+            } finally {
+                try {
+                    value.close();
+                } catch (IOException e) {
+                    log.error("error when close output stream {}", key);
+                }
+            }
+
+            needMoveFiles.put(key, getTargetLocation(key));
+        });
+    }
+
+    private FSDataOutputStream getOrCreateOutputStream(@NonNull String filePath) {
+        FSDataOutputStream fsDataOutputStream = beingWrittenOutputStream.get(filePath);
+        if (fsDataOutputStream == null) {
+            try {
+                fsDataOutputStream = FileSystemUtils.getOutputStream(filePath);
+                beingWrittenOutputStream.put(filePath, fsDataOutputStream);
+            } catch (IOException e) {
+                log.error("can not get output file stream");
+                throw new RuntimeException(e);
+            }
```

Comment:
Same as above and please check other places.

Reply:
The close logic is added in finishAndCloseFile, until prepareCommit we should release resources.

Reply:
Only if the stream is not fetched properly will an exception be thrown here, so there is no need to explicitly close the stream here.